llama_toolchain/core/datatypes.py

```diff
@@ -188,3 +188,19 @@ class PackageConfig(BaseModel):
 the dependencies of these providers as well.
 """,
     )
+
+
+@json_schema_type
+class BuildConfig(BaseModel):
+    name: str
+    distribution: str = Field(
+        default="local", description="Type of distribution to build (adhoc | {})"
+    )
+    api_providers: Optional[str] = Field(
+        default_factory=list,
+        description="List of API provider names to build",
+    )
+    package_type: str = Field(
```

Comment:
can we just use `BuildType` here ?

Reply:
This `BuildConfig` class store the config args, I think might be better to align with what is passed from command args as raw str for `package_type`.  

Side note: Feel like we should even simplify this `BuildType` enum class a bit. 

```
class BuildType(Enum):
    container = "container"
    conda_env = "conda_env"

    def descriptor(self) -> str:
        return "docker" if self == self.container else "conda"
```